### DISCUSSION TITLE
Will there be an enhancement to support time evolution of channels?

### DISCUSSION THREAD

**Discussion by salbert83** [NVlabs/sionna#493] [General]
For example, if one were to test HARQ processes on a TDL channel, one would need to evolve the channel in time beyond the 14 OFDM symbols used for PUSCH 5G NR.   Is there any current effort for this type of enhancement?  Thanks!

**Comment by SebastianCa** [comment]
Hi @salbert83,
I've moved this thread to the discussions as it is not related to an issue with the Sionna code.

I'm not exactly sure about your experiment, but you can already simulate the time evolution of channels with Sionna. This is independent of the PUSCH configuration and supports longer sequences. Please see the [Time Domain Channel](https://nvlabs.github.io/sionna/api/channel.wireless.html#time-domain-channel) for further details.

**Reply by salbert83** [reply]
If I want to simulate an OFDM channel for a long period of time, then my understanding is that the proper way to implement is just increase the number of OFDM symbols as necessary.  Is this correct?  Thanks.

**Reply by SebastianCa** [reply]
Yes, that is one way to implement the experiment, assuming you want to maintain the same underlying TDL model.

**Comment by 1ywan** [comment]
I think this feature is useful, think about that many scenarios in NR needs feedback, not only mentioned HARQ, but also CQI, PMI, SRS, etc., which are important part we want to try in Neural style transceiver, such as AI for Link Adapter schedule, beam management and beamforming weight prediction.

these features all require the time evolution channel model,  for all these features are need to run channel multi times.

and I think time evolution is easy to implement, 
```py
        # Time steps
        sample_times = tf.range(num_time_steps, dtype=self.rdtype)\
            /sampling_frequency
        sample_times = tf.expand_dims(insert_dims(sample_times, 6, 0), -1)
```
TDL use above code to generate sample_times, if we add a timing offset here, and if we can guarantee run channel in the same random seed, time evolution is done.
